test/integration/targets/deprecations/library/removeoption.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+
+DOCUMENTATION = r'''
+---
+module: removeoption
+short_description: noop
+description: does nothing, test for removal of option
+options:
+  one:
+    description:
+    - first option
+    type: bool
+    default: no
+  two:
+    description:
+    - second option
+    deprecated:
```

Comment:
They were allowed since deprecations were added for docs, but I don't think it has been used more than 1 or 2 times. I'll update schema too, not sure why that was disabled there, since 'per option deprecation' existed for modules much longer than for plugins.

Reply:
most of the time we just used aliases to rename an option and keep old name as alias/backwards compat, I don't think we've removed an option from a module since 1.0

Reply:
Option deprecations for modules were allowed, but documenting them with `deprecated` never was so far.

> most of the time we just used aliases to rename an option and keep old name as alias/backwards compat, I don't think we've removed an option from a module since 1.0

I think this happened quite a lot, especially in collections, but also in ansible/ansible. Actually it even happened in ansible-core: 9f82fc204bfa79a52049b4cbb45ae7196d329166 Some removals from Ansible 2.9: 808bf02588febe08f109364f20ad5d4a96a28100 88f0c8522882467d512eb4f1769e0eaf78404760

Reply:
Strict validation of module option docs (which effectively disallowed `deprecated` in it) was added in 04e816e13b4f3adda09a9a0f20266349bf89c870, which is now there for a bit over 7 years. That's more than half of the 12 years you mentioned.